lib/utils/getTopicThumbnail.ts

```diff
@@ -0,0 +1,36 @@
+import javaScript from "../../img/topicThumbnails/javascript.svg";
+import python from "../../img/topicThumbnails/python.svg";
+import typeScript from "../../img/topicThumbnails/typescript.svg";
+import java from "../../img/topicThumbnails/java.svg";
+import cSharp from "../../img/topicThumbnails/csharp.svg";
+import cpp from "../../img/topicThumbnails/cpp.svg";
+import php from "../../img/topicThumbnails/php.svg";
+import c from "../../img/topicThumbnails/c.svg";
+import ruby from "../../img/topicThumbnails/ruby.svg";
+import ai from "../../img/topicThumbnails/ai.svg";
+import ml from "../../img/topicThumbnails/ml.svg";
+import react from "../../img/topicThumbnails/react.svg";
+import contextThumbnailImage from "../../img/open-sourced-with-bg-icon.png";
+
+const topicThumbnails = {
```

Comment:
This should be synchronized with the array of topics returned in `utils/getTopicOptions.ts`. If we add another pre-defined topic, it should give an error if we forgot to add a corresponding image & vice versa.

Reply:
@Deadreyo  Yes, we should somehow synchronize this with the topic, and interest. But that's should be done in a different PR.
But at this moment this is not gonna throw any error if we add any predefined topic and do not add a corresponding image. Instead, it will show the Open Suaced logo.

Reply:
IMO should be done in this, since you're already at it. Not signifcant to require a separate PR.

Reply:
To clarify it more for you, since in `getTopicOptions` we have an array of the topics, and in your map you have the same topics as keys, you can look up the TS syntax that will generate a type out of the array that you can pass to the map.

Reply:
@Deadreyo  IMHO, what you're saying may not be feasible. While it's possible to generate a type from a one-dimensional array, this type cannot be used for an object. The elements of an array are string elements, whereas the elements of the object are acting like key-value pairs. Therefore, using the type of an array element for an object may not be appropriate.

We can get type out an array but that will just specify if the type of value is a string, number, or something. But here we do not need to know whether the interest array's elements are strings or numbers or not.

We need to know which elements are there on the interest list! The type of array is not gonna tell us that.

The interface might help here but maybe we need to have the same object structure for both `interests` and `thumbnails`

Reply:
It's actually possible:

```
const interests = ["javascript", "python", "java", "typescript", "rust", "csharp", "cpp", "php", "c", "ruby" , "ai", "ml", "react"] as const;

let map: Record<typeof interests[number], string> = {
  javascript: "javascript",
  ai: "ai",
  python: "python",
  java: "java",
  typescript: "typescript",
  rust: "rust",
  csharp: "csharp",
  cpp: "cpp",
  php: "php",
  c: "c",
  ruby: "ruby",
  react: "react",
  ml: "ml",
};```

Reply:
You can create the type `typeof interests[number]` in `getInterestOptions.ts` and export it to use it outside. Needless to mention, you'll also need to move interests array out of the function to the global scope to export it.

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type For more info

Reply:
@Deadreyo it works! Thanks!!